### PR TITLE
OWLS-86409: Additional Managed Server LifeCycle Tests and update usecases without Admin Server  

### PR DESCRIPTION
**Pull request by anpanigr**
Added the following new test cases
   ItServerStartPolicy.testStandaloneManagedRestart 
   ItServerStartPolicy.testConfiguredServerLifeCycleWithoutAdmin  ( msi mode ) 
   ItServerStartPolicy.testDynamicServerLifeCycleWithoutAdmin       (msi mode)

Added extra verification for starting/stopping a domain/cluster/servers with already started/stopped domain/cluster/servers

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3322/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java [hunk 1]
```diff
@@ -505,88 +458,73 @@ public void testDomainRestart() {
...
   /**
-   * The domain custom resource has a second configured manged server with serverStartPolicy IF_NEEDED
-   * Initially, the server will not come up since the replica count is set to 1
-   * Update the serverStartPolicy for config-cluster-server2 to ALWAYS
-   * by patching the resource definition with 
+   * Verify ALWAYS serverStartPolicy (config cluster) overrides replica count.
+   * The configured cluster has a second managed server(config-cluster-server2)
+   * with serverStartPolicy set to IF_NEEDED. Initially, the server will not 
+   * come up since the replica count for the cluster is set to 1. 
+   * Update the serverStartPolicy for the server config-cluster-server2 to i
```
Comment (alai8) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java:465]: extra "i' on this line

**Review by alai8 — APPROVED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java [hunk 2]
```diff
@@ -865,66 +845,173 @@ public void testStartManagedServerBeyondMaxClusterLimit() {
...
+   * Once the admin server is stopped, the managed server can only be started 
+   * in MSI (managed server independence) mode. To start a manged server in 
+   * MSI mode, the pre-requisite is the that the manged server MUST be started 
+   * once before admin server is shutdown, so that the embedded LDAP server is 
+   * replicated from admin server to the managed server. 
+   * In this case of MII and DomainInImage model, the server state/configuration
+   * is not saved once the server is shutdown unless we use domain-on-pv model
+   * So in MII case, startServer.sh script update the replica count but the 
+   * server startup is defered till we re-start the adminserver. Here the 
+   * operator tries to starti the managed server but it will keep on failing
```
Comment (hzhao-github) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java:950]: starti --> start

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java [hunk 3]
```diff
@@ -865,66 +845,173 @@ public void testStartManagedServerBeyondMaxClusterLimit() {
...
```
Comment (hzhao-github) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java:872]: starti --> start

**Comment by hzhao-github** [comment]
LGTM

**Review by hzhao-github — APPROVED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java (hunk 1 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java]: Fixed the typo

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java (hunk 2 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java]: Fixed the typo

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java (hunk 3 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java]: Fixed the typo

**Pull request merged**